src/Security/Authentication/Twitter/src/TwitterHandler.cs

```diff
@@ -318,5 +322,31 @@ private string ComputeSignature(string consumerSecret, string tokenSecret, strin
                 return Convert.ToBase64String(hash);
             }
         }
+
+        private async Task EnsureTwitterRequestSuccess(HttpResponseMessage response)
+        {
+            if (!response.IsSuccessStatusCode)
+            {
+                // Failure, attempt to parse Twitters error message
+                var errorContentStream = await response.Content.ReadAsStreamAsync();
```

Comment:
Remember to check the response content-type before attempting to deserialize as JSON. You want to avoid a secondary failure caused by trying to parse an empty or non-json response as json.